runtime/jvm/src/main/kotlin/kotlinx/serialization/Serialization.kt

```diff
@@ -35,12 +35,24 @@ actual fun <T: Any, E: T?> ArrayList<E>.toNativeArray(eClass: KClass<T>): Array<
 
 @Suppress("UNCHECKED_CAST")
 internal fun <T> Class<T>.invokeSerializerGetter(vararg args: KSerializer<Any>): KSerializer<T>? {
-    val companion: Any = try {
-        this.getField("Companion").apply { isAccessible = true }.get(null)
-    } catch (e: NoSuchFieldException) {
-        return null
-    }
-    return companion.javaClass.methods
-            .find {it.name == "serializer" && it.parameterTypes.size == args.size && it.parameterTypes.all {it == KSerializer::class.java } }
+    var serializer: KSerializer<T>? = null
+
+    // Search for serializer defined on companion object.
+    val companion = declaredFields.singleOrNull {it.name == "Companion"}?.apply {isAccessible = true}?.get(null)
+    if (companion != null)
+    {
```

Comment:
Please format braces around `if` according whole codestyle (and braces around lambdas too)